tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -90,13 +90,28 @@ private static void LogMatchesIfDebugEnabled(IReadOnlyCollection<object>? result
             if (additiveContext != null)
             {
                 // run the WAF and execute the results
-                result = additiveContext.Run(args, _security.Settings.WafTimeoutMicroSeconds);
+                if (runWithEphemeral)
+                {
+                    result = additiveContext.RunWithEphemeral(args, _security.Settings.WafTimeoutMicroSeconds);
+                }
+                else
+                {
+                    result = additiveContext.Run(args, _security.Settings.WafTimeoutMicroSeconds);
+                }
+
                 RecordTelemetry(result);
             }
         }
         catch (Exception ex) when (ex is not BlockException)
         {
-            Log.Error(ex, "Call into the security module failed");
+            if (logException is not null)
+            {
+                logException(Log, ex);
```

Comment:
I'm not sure I understand why we'd want to pass this, as if it comes from the rasp module, there will only be the key/value of rasp in the dictionary anyway, so we could instead just log all key/values in dictionary always (maybe first level is enough)

Reply:
Yes, that would be helpful for all the error cases. Done!